chessboard/skoved/chessboard.go

```diff
@@ -0,0 +1,59 @@
+package chessboard
+
+// Declare a type named Rank which stores if a square is occupied by a piece - this will be a slice of bools
+type Rank []bool
+
+// Declare a type named Chessboard contains a map of eight Ranks, accessed with values from "A" to "H"
+type Chessboard map[string]Rank
+
+// validateFile checks if an int is a valid chessboard file (1-8)
+func validateFile(file int) bool {
+	return file >= 1 && file <= 8
```

Comment:
This could count as an earlier refactoring. The function is only used one. I like it, but I think that in this case is not necessary.

Reply:
I wrote this thinking that it would get used again, but it did not :(